java/src/test/java/com/mlt/earcut/EarCutTriangulationTest.java

```diff
@@ -0,0 +1,158 @@
+package com.mlt.earcut;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.mlt.converter.triangulation.TriangulationUtils;
+import com.mlt.converter.triangulation.VectorTileConverter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import org.junit.jupiter.api.Test;
+import org.locationtech.jts.geom.*;
+import org.locationtech.jts.util.Assert;
+
+class EarcutTriangulationTest {
+  @Test
+  void triangulateVectorTile() throws IOException {
+    var path = Paths.get("../test/fixtures/amazon/5_5_11.pbf").toAbsolutePath();
+
+    var polygonConverter = new VectorTileConverter(path);
+
+    Assert.isTrue(polygonConverter.getNumTrianglesPerPolygon().size() > 0);
+    Assert.isTrue(polygonConverter.getIndexBuffer().size() > 0);
+
+    logVectorizationInformation(polygonConverter);
+  }
+
+  @Test
+  void triangulateVectorTile2() throws IOException {
+    var path = Paths.get("../test/fixtures/omt/13_4266_5468.mvt").toAbsolutePath();
+    var encodedTile = Files.readAllBytes(path);
+
+    var polygonConverter = new VectorTileConverter(encodedTile);
+
+    Assert.isTrue(polygonConverter.getNumTrianglesPerPolygon().size() > 0);
+    Assert.isTrue(polygonConverter.getIndexBuffer().size() > 0);
+    logVectorizationInformation(polygonConverter);
+  }
+
+  @Test
+  void verifyPolygonTriangulation() {
+    Polygon polygon = mock(Polygon.class);
+
+    when(polygon.getCoordinates()).thenReturn(getQuadraticPolygonCoordinates());
+    var triangulatedPolygon = TriangulationUtils.triangulatePolygon(polygon);
+
+    Assert.isTrue(triangulatedPolygon.getNumTrianglesPerPolygon() == 2);
+    Assert.isTrue(triangulatedPolygon.getIndexBuffer().size() == 6);
+  }
+
+  @Test
+  void verifyMultiPolygonTriangulation() {
+    MultiPolygon multiPolygon = mock(MultiPolygon.class);
+    Polygon outerPolygon = mock(Polygon.class);
+    when(outerPolygon.getCoordinates()).thenReturn(getQuadraticPolygonCoordinates());
+
+    Polygon innerPolygon = mock(Polygon.class);
+    Coordinate[] innerPolygonCoordinates = new Coordinate[4];
+    innerPolygonCoordinates[0] = new Coordinate(0.25, 0.25);
+    innerPolygonCoordinates[1] = new Coordinate(0.75, 0.25);
+    innerPolygonCoordinates[2] = new Coordinate(0.75, 0.75);
+    innerPolygonCoordinates[3] = new Coordinate(0.25, 0.75);
+
+    when(innerPolygon.getCoordinates()).thenReturn(innerPolygonCoordinates);
+
+    when(multiPolygon.getNumGeometries()).thenReturn(2);
+    when(multiPolygon.getGeometryN(0)).thenReturn(outerPolygon);
+    when(multiPolygon.getGeometryN(1)).thenReturn(innerPolygon);
```

Comment:
I'd suggest just creating the actual MultiPolygon/Polygon classes instead of using mocks.  One easy way is to use `new WKTReader().read("POLYGON ((30 10, 40 40, 20 40, 10 20, 30 10))")` to decode the well-known-text representation of a geometry